### PR TITLE
Profile wizard style consistency updates

### PR DESCRIPTION
**Pull request by samueljseay**
This is a rebase of #5019 I felt it was safer to create a new branch since they're not my changes.

I've tested the changes and they look good for me locally but @jameskoster could you please pull this down and test it. @joshuatf perhaps you should take a look as well.

I've updated the snapshot test because the rendering has changed (I think maybe we shouldn't have snapshot tests on internal components anyway because they tend to churn a lot).

To test I focussed most of my time around the monthly pricing toggle and all the styling in the profile wizard.

If this looks OK, then we can close #5019

### CODE REVIEW

File: client/profile-wizard/steps/product-types/index.js [hunk 1]
```diff
@@ -100,15 +105,17 @@ export class ProductTypes extends Component {
...
-					{ __( 'Choose any that apply' ) }
-				</H>
+				<div className="woocommerce-profile-wizard__step-header">
+					<Text variant="title.small" as="h2">
+						{ __(
+							'What type of products will be listed?',
+							'woocommerce-admin'
+						) }
+					</Text>
+					<Text vairant="body">
```
Comment (becdetat) [inline client/profile-wizard/steps/product-types/index.js:115]: Typo here

Comment (samueljseay) [reply client/profile-wizard/steps/product-types/index.js:115]: wow thanks for spotting this.

Comment (samueljseay) [reply client/profile-wizard/steps/product-types/index.js]: Fixed here: https://github.com/woocommerce/woocommerce-admin/pull/5059/commits/232b98c1e2656973e4654ed562b72592d9f52763

**Review by joshuatf — COMMENTED** [review]
Changes look good to me.  Will leave final ✅ to @jameskoster

**Comment by joshuatf** [comment]
>I've updated the snapshot test because the rendering has changed (I think maybe we shouldn't have snapshot tests on internal components anyway because they tend to churn a lot).

Ya, I think you're right.

**Review by jameskoster — APPROVED** [review]
:shipit:

**Pull request merged**